tests/by-util/test_paste.rs

```diff
@@ -183,13 +186,174 @@ fn test_delimiter_list_ending_with_escaped_backslash() {
 fn test_delimiter_list_ending_with_unescaped_backslash() {
     for d in ["-d", "--delimiters"] {
         new_ucmd!()
-            .args(&[d, "\\"])
+            .args(&[d, r"\"])
             .fails()
-            .stderr_contains("delimiter list ends with an unescaped backslash: \\");
+            .stderr_contains(r"delimiter list ends with an unescaped backslash: \");
+
+        new_ucmd!()
+            .args(&[d, r"\\\"])
+            .fails()
+            .stderr_contains(r"delimiter list ends with an unescaped backslash: \\\");
+
+        new_ucmd!()
+            .args(&[d, r"_\"])
+            .fails()
+            .stderr_contains(r"delimiter list ends with an unescaped backslash: _\");
+    }
+}
+
+#[test]
+fn test_delimiter_list_empty() {
+    for option_style in ["-d", "--delimiters"] {
+        new_ucmd!()
+            .args(&[option_style, "", "-s"])
+            .pipe_in(
+                "\
+A ALPHA 1 _
+B BRAVO 2 _
+C CHARLIE 3 _
+",
+            )
+            .succeeds()
+            .stdout_only(
+                "\
+A ALPHA 1 _B BRAVO 2 _C CHARLIE 3 _
+",
+            );
+    }
+}
+
+// Was panicking (usize subtraction that would have resulted in a negative number)
+// Not observable in release builds, since integer overflow checking is not enabled
+#[test]
+fn test_delimiter_truncation() {
+    for option_style in ["-d", "--delimiters"] {
+        new_ucmd!()
+            .args(&[option_style, "!@#", "-s", "-", "-", "-"])
+            .pipe_in(
+                "\
+FIRST
+SECOND
+THIRD
+FOURTH
+ABCDEFG
+",
+            )
+            .succeeds()
+            .stdout_only(
+                "\
+FIRST!SECOND@THIRD#FOURTH!ABCDEFG
+
+
+",
+            );
+    }
+}
+
+#[test]
+fn test_non_utf8_input() {
+    // 0xC0 is not valid UTF-8
+    const INPUT: &[u8] = b"Non-UTF-8 test: \xC0\x00\xC0.\n";
+
+    new_ucmd!()
+        .pipe_in(INPUT)
+        .succeeds()
+        .stdout_only_bytes(INPUT);
+}
```

Comment:
Great improvement compared to the previous version with all its consts :)

Reply:
I had to discover "byte strings" (https://doc.rust-lang.org/stable/reference/tokens.html#characters-and-strings)!